File: spec/test_lexer_result_value.c
#include "testparser.h"
#include <assert.h>
#include <string.h>

int main()
{
    char const * input = "x";
    p_context_t * context;
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    assert(p_result(context) == 1u);
    p_context_delete(context);

    input = "fabulous";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    assert(p_result(context) == 8u);
    p_context_delete(context);

    return 0;
}

File: spec/test_lexer_result_value.rs
use testparser::*;

fn main() {
    let mut context = p_context_new(b"x");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    assert_eq!(1, p_result(&context));
    p_context_delete(context);

    let mut context = p_context_new(b"fabulous");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    assert_eq!(8, p_result(&context));
    p_context_delete(context);
}
